crates/primitives/src/transaction/pooled.rs

```diff
@@ -78,7 +142,39 @@ impl Encodable for PooledTransactionsElement {
     /// Encodes an enveloped post EIP-4844 [PooledTransactionsElement].
     fn encode(&self, out: &mut dyn bytes::BufMut) {
         match self {
-            Self::Transaction(tx) => tx.encode(out),
+            Self::Legacy { transaction, signature, hash } => {
+                // construct signed transaction
+                let signed_tx = TransactionSigned {
+                    transaction: Transaction::Legacy(transaction.clone()),
+                    signature: *signature,
+                    hash: *hash,
+                };
+
+                // encode signed transaction
+                signed_tx.encode(out);
```

Comment:
we're now using clones here, this is way too expensive.

This can be solved by using 

https://github.com/paradigmxyz/reth/blob/c62a07de80ea3a81283fdbd4e8b26e1ff4570bf0/crates/primitives/src/transaction/mod.rs#L373-L396 

directly

Reply:
right, I had even refactored those so it could be used easily with the inner tx, just forgot to use it, will add in a followup